include/unifex/sync_wait.hpp

```diff
@@ -113,16 +115,25 @@ template <typename Result, typename Sender>
 std::optional<Result> _impl(Sender&& sender) {
   using promise_t = _sync_wait::promise<Result>;
   promise_t promise;
-  manual_event_loop ctx;
+  auto blockingResult = blocking(sender);
+  const bool completesInline = blockingResult == blocking_kind::always_inline;
+
+  manual_lifetime<manual_event_loop> ctx;
```

Comment:
This is not exception-safe and will leak in the case that `connect()` below throws.
You should be able to just use a `std::optional` and just call `.emplace()` here instead.

Reply:
Quite right, I was expecting this to come up. For some reason I didn't think of emplace when I was playing with optional.